engine/consensus/provider/engine.go

```diff
@@ -77,75 +80,49 @@ func (e *Engine) Done() <-chan struct{} {
 	return e.unit.Done()
 }
 
-// SubmitLocal submits an event originating on the local node.
-func (e *Engine) SubmitLocal(event interface{}) {
+// ProvideProposal asynchronously submits our proposal to all non-consensus nodes.
+func (e *Engine) ProvideProposal(proposal *messages.BlockProposal) {
 	e.unit.Launch(func() {
-		err := e.ProcessLocal(event)
+		err := e.broadcastProposal(proposal)
 		if err != nil {
-			engine.LogError(e.log, err)
+			// TODO: once error handling in broadcastProposal is updated, this can ctx.Throw instead of logging
+			e.log.Err(err).Msg("unexpected error while broadcasting proposal")
 		}
 	})
 }
 
-// Submit submits the given event from the node with the given origin ID
-// for processing in a non-blocking manner. It returns instantly and logs
-// a potential processing error internally when done.
-func (e *Engine) Submit(channel network.Channel, originID flow.Identifier, event interface{}) {
-	e.unit.Launch(func() {
-		err := e.Process(channel, originID, event)
-		if err != nil {
-			engine.LogError(e.log, err)
-		}
-	})
-}
-
-// ProcessLocal processes an event originating on the local node.
-func (e *Engine) ProcessLocal(event interface{}) error {
-	return e.unit.Do(func() error {
-		return e.process(e.me.NodeID(), event)
-	})
-}
-
-// Process processes the given event from the node with the given origin ID in
-// a blocking manner. It returns the potential processing error when done.
+// Process logs a warning for all received messages, as this engine's channel
+// is send-only - no inbound messages are expected.
 func (e *Engine) Process(channel network.Channel, originID flow.Identifier, event interface{}) error {
-	return e.unit.Do(func() error {
-		return e.process(originID, event)
-	})
+	e.log.Warn().
+		Str("channel", channel.String()).
+		Str("origin_id", originID.String()).
+		Msgf("received unexpected message (%T), dropping...", event)
+	return nil
 }
 
-// process processes the given ingestion engine event. Events that are given
-// to this function originate within the provider engine on the node with the
-// given origin ID.
-func (e *Engine) process(originID flow.Identifier, event interface{}) error {
-	switch ev := event.(type) {
-	case *messages.BlockProposal:
-		return e.onBlockProposal(originID, ev)
-	default:
-		return fmt.Errorf("invalid event type (%T)", event)
+// broadcastProposal is used when we want to broadcast a local block to the rest  of the
+// network (non-consensus nodes). We broadcast to consensus nodes in compliance engine.
+// TODO error handling - differentiate expected errors in Snapshot
+func (e *Engine) broadcastProposal(proposal *messages.BlockProposal) error {
+
+	// sanity check: we should only broadcast proposals that we proposed here
+	if e.me.NodeID() != proposal.Header.ProposerID {
+		return fmt.Errorf("sanity check failed: attempted to provide another node's proposal (%x!=%x)", e.me.NodeID(), proposal.Header.ProposerID)
```

Comment:
this sanity check is also in the compliance Engine: https://github.com/onflow/flow-go/blob/f7b85fbb7c67df694a8b92aa4dc7d96860e54702/engine/consensus/compliance/engine.go#L454-L457
do we need it in both places? 
... the check is probably cheap enough that it makes no practical difference, so why not 🤷

Reply:
Yeah I'm more inclined to just leave it